src/common/lib/transport/connectionmanager.ts

```diff
@@ -916,7 +916,13 @@ class ConnectionManager extends EventEmitter {
    *********************/
 
   getError(): IPartialErrorInfo | string {
-    return this.errorReason || this.getStateError();
+    if (this.errorReason) {
+      // create new PartialErrorInfo so it has the correct stack trace
+      // which points to the place which caused us to return this error.
+      return PartialErrorInfo.fromValues(this.errorReason);
```

Comment:
Would it be better to wrap the existing error as the `cause` of the returned error, so that you have both stack traces? (not saying it would be, just interested to know your thoughts)

Reply:
It didn't feel like it would provide any useful additional information here.
To me, it seems that you would want to use the `cause` field in situations where you are `try-catch`ing some expected error and then wrapping it to add more contextual information about what happened. In this specific case, we just want to create a new instance of `PartialErrorInfo` to get the correct stack trace; we don't have any additional context to provide.

Reply:
well i was imagining the additional context in this case being where the connection error originated. for example, if you're trying to attach to a channel when your connection is closed, the error would tell you where you tried to attach to the channel, and the error.cause would tell you where you closed the connection - both useful bits of information for debugging.

Reply:
Well, I guess there is no harm in adding that info. Currently, though, only Firefox displays the `cause` in the console ([source](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause#browser_compatibility)), so it wouldn't be much help by default for everyone, as devs would need to manually try/catch the error and access the `cause` property in other browsers. But at this point, yeah, it's just another bit of information that might help, I guess?
Added `cause` [here](https://github.com/ably/ably-js/compare/eda9c4195e8522cae3ebb37e15d5a59176e05865..f15aca3cc2b9d2f1c253aa7ff3fc87154434721c)

Reply:
we actually override `Error.cause` in `ErrorInfo` with our own `cause` field so i'm not sure if that will apply, but yeah just having the information accessible will be useful either way, if someone raises an issue we can tell them to check `Error.cause` rather than having to look through their code and find where the connection error originates from

Reply:
We don't do anything special with the `cause` field in `ErrorInfo` classes, so in the end it's just a regular `Error.cause` property by js spec. So it's not displayed in chromimum browsers and safari, but on firefox it looks like this:
![image](https://github.com/ably/ably-js/assets/9639405/f9c3a063-8f18-4eed-8034-71ee45aeefff)